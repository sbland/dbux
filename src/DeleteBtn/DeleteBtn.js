import React, { useState, useEffect, Fragment } from 'react';
import PropTypes from 'prop-types';

const DeleteBtn = ({ handleConfirmedDelete, object, className, disabled, notes, PopupPanel }) => {
  const [requestDelete, setRequestDelete] = useState(false);
  const [areYouSure, setAreYouSure] = useState(false);

  useEffect(() => {
    if (areYouSure && requestDelete) {
      handleConfirmedDelete();
      setAreYouSure(false);
      setRequestDelete(false);
    }
  }, [areYouSure, requestDelete, handleConfirmedDelete]);

  return (
    <>
      <PopupPanel isOpen={requestDelete && !areYouSure} handleClose={() => setRequestDelete(false)}>
        <div>
          <h1>Are You Sure?</h1>
          <div>
            <button
              type="button"
              className="areYouSureCancel button-one"
              onClick={() => setRequestDelete(false)}
            >
              Cancel
            </button>
            <button
              type="button"
              className="areYouSureAccept button-two"
              onClick={() => setAreYouSure(true)}
            >
              Delete {object}
            </button>
          </div>
          {notes && <div>{notes}</div>}
        </div>
      </PopupPanel>
      <button
        disabled={disabled}
        onClick={() => setRequestDelete(true)}
        type="button"
        className={`deleteBtn ${className}`}
      >
          <span className="emoji" role="img" aria-label="Delete">
            üóëÔ∏è
          </span>
      </button>
    </>
  );
};

DeleteBtn.propTypes = {
  handleConfirmedDelete: PropTypes.func.isRequired,
  object: PropTypes.string.isRequired,
  className: PropTypes.string,
  disabled: PropTypes.bool,
  notes: PropTypes.string,
};

DeleteBtn.defaultProps = {
  className: 'button-one',
  disabled: false,
  notes: '',
};

export default DeleteBtn;
