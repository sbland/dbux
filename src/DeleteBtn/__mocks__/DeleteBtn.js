import React from 'react';


const DeleteBtn = ({

}) => (
  <div className="DeleteBtn">
    Mock Delete Btn
  </div>
);

export default DeleteBtn;
