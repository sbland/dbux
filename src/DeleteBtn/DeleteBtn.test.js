import React from 'react';
import { shallow, mount } from 'enzyme';
import DeleteBtn from './DeleteBtn';

const handleConfirmedDelete = jest.fn();
const defaultProps = {
  handleConfirmedDelete,
  object: 'Item',
  className: 'demoClass',
  disabled: false,
  PopupPanel: ({children}) => children
};

describe('DeleteBtn', () => {
  it('Renders', () => {
    shallow(<DeleteBtn {...defaultProps} />);
  });
  it('Matches Snapshot', () => {
    const component = shallow(<DeleteBtn {...defaultProps} />);
    const tree = component.debug();
    expect(tree).toMatchSnapshot();
  });
  describe('Unit Tests', () => {
    let component;
    beforeEach(() => {
      component = mount(<DeleteBtn {...defaultProps} />);
    });
    describe('are you sure process', () => {
      const clickDeleteBtn = (c) => {
        const deleteBtn = c.find('.deleteBtn');
        deleteBtn.simulate('click');
      };
      test('should open are you sure panel when delete button pressed', () => {
        let areYouSurePanel = component.find(PopupPanel);
        expect(areYouSurePanel.props().isOpen).toEqual(false);
        clickDeleteBtn(component);
        areYouSurePanel = component.find(PopupPanel);
        expect(areYouSurePanel.props().isOpen).toEqual(true);
      });
      test('should call confirm delete id we click accept button in are you sure panel', () => {
        clickDeleteBtn(component);
        let areYouSurePanel = component.find(PopupPanel);
        const areYouSureAcceptBtn = areYouSurePanel.find('.areYouSureAccept');
        areYouSureAcceptBtn.simulate('click');
        areYouSurePanel = component.find(PopupPanel);
        expect(areYouSurePanel.props().isOpen).toEqual(false);
        expect(handleConfirmedDelete).toHaveBeenCalledWith();
      });
    });
  });
});
