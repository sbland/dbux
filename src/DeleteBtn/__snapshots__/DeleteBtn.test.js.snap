// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DeleteBtn Matches Snapshot 1`] = `
"<Fragment>
  <PopupPanelMock isOpen={false} handleClose={[Function: handleClose]}>
    <div>
      <h1>
        Are You Sure?
      </h1>
      <div>
        <button type=\\"button\\" className=\\"areYouSureCancel button-one\\" onClick={[Function: onClick]}>
          Cancel
        </button>
        <button type=\\"button\\" className=\\"areYouSureAccept button-two\\" onClick={[Function: onClick]}>
          Delete 
          Item
        </button>
      </div>
    </div>
  </PopupPanelMock>
  <button disabled={false} onClick={[Function: onClick]} type=\\"button\\" className=\\"deleteBtn demoClass\\">
    <Emoji emoj=\\"🗑️\\" label=\\"Delete\\" />
  </button>
</Fragment>"
`;
